client/components/common/BlockForm.vue

```diff
@@ -61,7 +59,10 @@ export default {
   },
   created() {
     this.values = Object.fromEntries(this.fields.map(f => [f.id, f.value]));
-    this.validators = Object.fromEntries(this.fields.map(f => [f.id, f.optional ? (v => '') : (v => v ? '' : 'required field')]));
+    this.fields.forEach(f => {
+      const validators = [f.optional ? (v => '') : (v => v ? '' : 'required field'), this.validators[f.id] || (v => '')];
+      this.$set(this.validators, f.id, v => validators.map(validator => validator(v)).find(x => x));
+    });
```

Comment:
I got lost here

Reply:
ah yeah this maybe deserves a comment. Basically that just returns the message of the first validator that returned something non-empty.

Reply:
Just remembered that I ended up replacing this with a reduction because it is a bit more readable, so won't need a comment since this doesn't exist anymore.